### PR TITLE
Curve 3pool

### PR DESCRIPTION
**Pull request by chanhosuh**
## Description
Cortex DAO (formerly APY.Finance) requires a USD price feed for the Curve 3Pool LP token (aka 3CRV).  This feed will be used to price deposits made in 3CRV to ensure the correct amount of our share tokens are minted.  

## Changes

For the pricing methodology, we have chosen the USD liquidation value of a 3CRV token.  This is the aggregate USD value of the stablecoins redeemed from a single 3CRV token using the 3Pool's standard redemption function `remove_liquidity`; there are other ways of redeeming that incur slippage so decrease value.

The methodology is explained in detail (working from the 3Pool's vyper contract) in the README.

_Technical note:_ Note the 3Pool (like all Curve pools) have a `virtual_price` value, but this does not reflect USD value and is an internal tracking method to properly allocate fees to LPs.  For example, virtual price never decreases unless the pool's A parameter is adjusted; in particular, even if all stables in the 3Pool depegged to $0.50 the virtual price is unaffected.

The adapter is a composite, feeding allocations (one for each of DAI, USDC, and Tether) into the Token Allocation adapter, which pulls the USD price for each stable and aggregates the values.  We believe that pulling a real-time off-chain value for each stable simultaneously like this is the correct approach to avoid arbitrage of the 3CRV token.

The adapter code results from reworking the code from `composites/apy-finance`.  Hopefully this didn't result in unclean code and satisfies Chainlink standards.  I've created a similar test as for that adapter and added another one.  Please let me know what additional testing may be required or how tests or coding can be improved.

## Steps to Test

From the repo root `yarn test composites/curve-3pool/test/integration` will run the tests.

For ad-hoc testing (and generating the sample output), I spun up the `coingecko-adapter` and `curve-3pool-adapter` in docker containers.

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [ ] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)
- [x] All code changes have 100% unit and integration test coverage. If testing is not applicable or too difficult to justify doing, the reasoning should be documented explicitly in the PR.

### CODE REVIEW

File: packages/composites/curve-3pool/CHANGELOG.md
Comment (austinborn) [inline packages/composites/curve-3pool/CHANGELOG.md:3]: Please delete this file since these are from `apy-finance`'s changelog

File: packages/composites/curve-3pool/test/integration/fixtures.ts
Comment (austinborn) [inline packages/composites/curve-3pool/test/integration/fixtures.ts:96]: If you're returning the same value every time for the same inputs, you can use `.persist()` and remove the other copies of the post/reply.

**Review by austinborn — CHANGES_REQUESTED** [review]
Sorry for the wait, but this looks great! Only modification necessary is to remove the CHANGELOG since that only applies for the `api-finance` EA.

**Comment by austinborn** [comment]
Actually 2 more things:
1) You should be able to add a changeset for the creation of the new EA with `yarn changeset`. The interface for this command should be straightforward: type the name to narrow the filter, `space` to select, `enter` to move to the next screen. This should be a `major` bump since it's a new EA.
2) Add this EA to `packages/tsconfig.json`.

File: packages/composites/curve-3pool/test/integration/fixtures.ts
Comment (chanhosuh) [reply packages/composites/curve-3pool/test/integration/fixtures.ts:96]: Tbh, not familiar with nock, but I have two different interceptors for each post actually (just $1 for the first set and slightly off $1 for the next), so I think this is fine?

**Comment by chanhosuh** [comment]
I think I addressed the comments.  I was able to generate a new changeset, but I probably buggered the `.pnp.cjs` and `yarn.lock` somehow.

File: packages/composites/curve-3pool/test/integration/fixtures.ts
Comment (austinborn) [reply packages/composites/curve-3pool/test/integration/fixtures.ts:96]: Oh I see, yeah that's fine then

**Review by austinborn — DISMISSED** [review]

**Comment by austinborn** [comment]
Thank you for the updates, this looks good now 👍 . I'm wondering if you would be able to remove all of the `.yarn` package updates. Maybe try `yarn cache clean` and `git checkout .yarn`?

**Comment by chanhosuh** [comment]
> Thank you for the updates, this looks good now 👍 . I'm wondering if you would be able to remove all of the `.yarn` package updates. Maybe try `yarn cache clean` and `git checkout .yarn`?

Ok, sorry.  I had a conflicting version of node in the adapter's `package.json`.  That's been fixed and the `.yarn` files should be the same as at the base of the branch.

**Review by austinborn — APPROVED** [review]

**Comment by austinborn** [comment]
@chanhosuh Thanks for reviewing the `yarn` issue for me. I noticed that the PR checks have been stuck, so could you push an empty commit or re-push the last commit? We just need to change the commit history somehow to re-trigger the checks to run...

**Comment by chanhosuh** [comment]
> @chanhosuh Can you try pushing an empty commit here? The gh workflow got stuck but updating the commit history should jumpstart it.

Ok, pushed the empty commit.  Waiting for the checks to complete.  They seem to be taking a while.

**Comment by chanhosuh** [comment]
@austinborn not sure why the checks aren't running after I pushed an empty commit.  Also, I pushed a trivial change to the README to see if that would work.  That unfortunately undid your review.

**Comment by austinborn** [comment]
@chanhosuh the branch is back out of date, so can you try merging the conflicts and pushing one more time? If it fails, I will try to open a fork of the PR myself and get it approved by another engineer on the team.

**Review by austinborn — DISMISSED** [review]

**Comment by austinborn** [comment]
@chanhosuh The branch is still out-of-date? You may need to pull the latest from the `remote develop` again...

**Comment by chanhosuh** [comment]
@austinborn updated the type imports etc; I think (most) checks should pass now.

**Comment by austinborn** [comment]
Great, re-running now

**Comment by chanhosuh** [comment]
@austinborn Everything passed except the soak test.  It failed on the AWS credentials step.  Is this something I should fix or something you guys need to handle?

**Review by austinborn — APPROVED** [review]

**Comment by austinborn** [comment]
@chanhosuh No worries about the soak test, it will not work since you are missing the correct secrets, but that check should not be required for external PRs (I believe). Looks like some changes were introduced since the last sync so you will need to re-sync, but I can re-approve if it dismisses my current approval.

**Comment by austinborn** [comment]
@chanhosuh No worries about the soak test, it will not work since you are missing the correct secrets, but that check should not be required for external PRs (I believe). Looks like some changes were introduced since the last sync so you will need to re-sync, but I can re-approve if it dismisses my current approval.

**Comment by chanhosuh** [comment]
@austinborn synced the PR again but the workflows need your approval.  Thanks

**Pull request merged**
